naucse/models.py

```diff
@@ -354,14 +354,30 @@ def _time(self, time):
             return datetime.datetime.combine(self.date, time)
         return None
 
+    def _session_time(self, key):
+        sesion_time = self.info.get('time')
+        if sesion_time:
+            return time_from_string(sesion_time[key])
+        return None
+
+    @reify
+    def has_custom_time(self):
+        return self._session_time('start') is not None
```

Comment:
Could you check end as well? I can imagine a session that's planned ot run a bit longer.